### PR TITLE
Add permission for throttling

### LINKED ISSUE #1255: Throttling is bothersome to some permission groups
Although related to #929;

I wish there was an option to enable throttling only for specific permission groups. As a moderator on discuss it sometimes block my processes to handle tasks efficiently when having to wait for the grace period.

### PR DESCRIPTION
**Pull request by datitisev**
**Fixes #1255**

**Changes proposed in this pull request:**
- Add permission for floodgate throttling

**Reviewers should focus on:**
- Not sure if the logic should be in the controllers, or somewhere in the handler... mainly because extensions will have to implement the permission themselves as well
- Should this be a bypass permission instead?
- Should the name be `floodgate.throttling`, simple `throttling`, or something else?

**Screenshot**
image

**Confirmed**

- [x] Frontend changes: tested on a local Flarum installation.
- [x] Backend changes: tests are green (run `php vendor/bin/phpunit`).

**Required changes:**

- [ ] Related core extension PRs: flarum/english (not done yet, waiting for confirmation)

### CODE REVIEW

**Comment by CDK2020** [comment]
I think it should be bypass. there will be a lot less people who will be allowed to bypass throttling then those who should be throttled. Not that this matters but the grammar of the function makes more sense. `$actor->can('throttle');` vs. `$actor->can('bypass');`

**Comment by franzliedke** [comment]
Thanks for the PR! We'll tackle this once beta.8 is released.

File: migrations/2018_07_21_000100_seed_default_group_permissions.php [hunk 1]
```diff
@@ -16,10 +16,11 @@
     // Guests can view the forum
     ['permission' => 'viewDiscussions', 'group_id' => Group::GUEST_ID],
 
-    // Members can create and reply to discussions, and view the user list
+    // Members can create and reply to discussions, view the user list, and be throttled
     ['permission' => 'startDiscussion', 'group_id' => Group::MEMBER_ID],
     ['permission' => 'discussion.reply', 'group_id' => Group::MEMBER_ID],
     ['permission' => 'viewUserList', 'group_id' => Group::MEMBER_ID],
+    ['permission' => 'floodgate.throttling', 'group_id' => Group::MEMBER_ID],
```
Comment (franzliedke) [inline migrations/2018_07_21_000100_seed_default_group_permissions.php]: Editing old, already-released migrations, is a no-go.

File: src/Api/Controller/CreateDiscussionController.php [hunk 2]
```diff
@@ -65,7 +65,7 @@ protected function data(ServerRequestInterface $request, Document $document)
         $actor = $request->getAttribute('actor');
         $ipAddress = array_get($request->getServerParams(), 'REMOTE_ADDR', '127.0.0.1');
 
-        if (! $request->getAttribute('bypassFloodgate')) {
+        if ($actor->can('floodgate.throttling') && ! $request->getAttribute('bypassFloodgate')) {
```
Comment (franzliedke) [inline src/Api/Controller/CreateDiscussionController.php]: I would turn this around as well - throttling should be on for everyone (the current default) unless explicitly disabled.

That also takes us to naming... what about something like `postWithoutThrottle`? That's the thing you are *permitted* to do.

**Comment by datitisev** [comment]
See #1938.

**Pull request closed without merging**